Extensions/AboutUsParser.js

```diff
@@ -0,0 +1,316 @@
+    Finalpath = Finalpath ? Finalpath : C.ComponentsPath;
+
+    const component = nestedComponent ? `:^) ${name} :::\r\n` : '';
+    let path = Path.join(Finalpath, language, `${fileName}.html`);
+
+    writeFile(path, `${component}${content}`, append_file);
+    console.log(`Successfully created component ${component},\r\nAppend File Mode: ${append_file}`);
+}
+
+/** Synchronously writes a string into a file or appends it
+ * 
+ * @param {String} path Path of the file
+ * @param {String} string String to write into the file
+ * @param {Boolean} appendFile 
+ * @returns {void}
+ */
+function writeFile(path, string, appendFile) {
+    switch (appendFile) {
+        case true:
+            fs.appendFileSync(path, `${string}\r\n`, { flag: 'a+', mode: 0666 });
+            break;
+        case false:
+            fs.writeFileSync(path, `${string}\r\n`, { flag: 'w+', mode: 0666 });
+            break;
+        default:
+            throw new Error('No valid action was chosen for writing a file!');
+    }
+}
+
+/** Creates a Json file with our data
+ * 
+ * @param {Array} array Array of `new Paragraph()` objects
+ */
+function saveAsJson(array) {
+    return fs.writeFileSync(Path.join(C.ModulePath, C.jsonFile), JSON.stringify(array, null, 4), {flag: 'w+', mode: 0666});
+}
+
+/** Loads a Json file and returns its content parsed
+ * 
+ * @param {String} file Path to the Json file 
+ * @returns {String} parsed Json file content
+ */
+function loadFromJsonFile(file) {
+    return JSON.parse(fs.readFileSync(file, { mode: 0666 }));
+}
+
+/** Creates component files from objects array
+ * 
+ * @param {Array} array Array of objects to be created as component
+ */
+function transpileArrayIntoComponents(array) {
+    var nested = array.length > 1;
+    array.map((section, sectionIndex) => {
+        section.content.map(content => {
+            let string = getStringFromArray(content.data);
+            if(sectionIndex == 0)
+                createComponent(C.ComponentsPath, section.folder, string, content.language, false, nested);
+            else
+                createComponent(C.ComponentsPath, section.folder, string, content.language, true, nested);
+        });
+    });
+}
+
+/**
+ * 
+ * @param {Array} array 
+ * @deprecated This will be added later on, it should check whether Json file contains objects with paths to photos and ask whether to combine them with newly parsed objects array of data
+ */
+function checkJsonForPhotos(parsedJson) {
+    let arr = parsedJson;
+
+    return parsedJson;
+}
+
+/** Counts the number of subdirectories in a directory
+ * 
+ * @param {String} dir Path to your directory where you'd like to count subdirectories
+ * @returns {Number} count of subdirectories
+ */
+function countSubDirs(dir) {
+    dir = Path.normalize(dir);
+    var parse = dir.split('\\'),
+        index = parse.indexOf(Path.parse(dir).base);
+    return Math.floor(parse.length - index);
+}
+
+function AboutUs() {
```

Comment:
I am really bothered by transpilation needing human attention to interact with console :disappointed: this makes it incredibly hard to automate. how can we replace this?

Reply:
No outcome of our discussion yet leaving this open, but it's unnecessary.